src/common/Themes/icon_helpers.cpp

```diff
@@ -14,7 +14,7 @@ HRESULT GetIconIndexFromPath(_In_ PCWSTR path, _Out_ int* index)
     if (!PathIsRelative(path))
     {
         DWORD attrib = GetFileAttributes(path);
-        HIMAGELIST himl = (HIMAGELIST)SHGetFileInfo(path, attrib, &shFileInfo, sizeof(shFileInfo), (SHGFI_SYSICONINDEX | SHGFI_SMALLICON | SHGFI_USEFILEATTRIBUTES));
```

Comment:
Why auto instead of a static cast here?

Reply:
The himl variable is only used to check If the call to SHGetFileInfo was successful.
According to [link](https://learn.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-shgetfileinfow#shgfi_sysiconindex-0x000004000) when passing a SHGFI_SYSICONINDEX flag the return should be a HANDLE to an image list (HIMAGELIST) as pointer should be 0 if unsuccessful.
I could change it to reinterpret_cast, but i felt no need for the cast.